ls1intum/Artemis
# `Programming exercises`: Fix an issue in which the submission policy was not enforced

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
The lock repository submission policy is currently not enforced.

### Description
This PR fixes the retrieval of the submission policy from the database. This means that the online editor in the client will show an error if a student pushes to their repository after they reached the submission limit. The correct locking of the client will be fixed in a follow-up PR.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming Exercise with a lock repository submission policy configured

1. Log in to Artemis as the student and go to the exercise with the submission policy.
2. Submit until you reached the submission limit (either via the online editor or via the local Git client).
3. Make sure you cannot push after the submission limit is reached.
4. Try to push to the student's repository as the instructor and make sure you are still able to do that.

#### Exam Mode Testing
- Repeat the steps from above for an exam exercise.

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [RepositoryProgrammingExerciseParticipationResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4821-3/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.repository/RepositoryProgrammingExerciseParticipationResource.html) | 80% | ok |
